.github/workflows/unittests.yml

```diff
@@ -8,52 +9,94 @@
      - '*'
    schedule:
      - cron: '59 23 * * *'
+   workflow_dispatch:
+    inputs:
+      version:
+        description: Manual CI Run
+        default: test
+        required: false
 
  jobs:
-   unittests:
-     name: CI (${{ matrix.os }}-${{ matrix.environment-file }})
+   tests:
+     name: ${{ matrix.os }}, ${{ matrix.environment-file }}
      runs-on: ${{ matrix.os }}
-     continue-on-error: false
-     timeout-minutes: 45
+     timeout-minutes: 30
      strategy:
        matrix:
-         os: [ubuntu-latest, macos-latest, windows-latest]
-         environment-file: [ .ci/39.yml, .ci/310.yml, .ci/311.yml]
+         os: [ubuntu-latest]
+         environment-file: [
+            ci/310.yml,
+            ci/311.yml,
+            ci/312.yml,
+         ]
+         include:
+           - environment-file: ci/312.yml
+             os: macos-latest
+           - environment-file: ci/312.yml
+             os: macos-14 # Apple Silicon
+           - environment-file: ci/312.yml
+             os: windows-latest
        fail-fast: false
 
+     defaults:
+       run:
+         shell: bash -l {0}
+
      steps:
        - name: checkout repo
-         uses: actions/checkout@v2
+         uses: actions/checkout@v4
+         with:
+           fetch-depth: 0 # Fetch all history for all branches and tags.
        
        - name: setup micromamba
-         uses: mamba-org/provision-with-micromamba@main
+         uses: mamba-org/setup-micromamba@v1
          with:
            environment-file: ${{ matrix.environment-file }}
            micromamba-version: 'latest'
         
-       - name: reinstall tobler - bash
-         shell: bash -l {0}
-         run: pip install -e . --no-deps --force-reinstall
-         if: matrix.os != 'windows-latest'
-      
-       - name: reinstall tobler - powershell
-         shell: powershell
-         run: pip install -e . --no-deps --force-reinstall
-         if: matrix.os == 'windows-latest'
+       ##############################################################################
```

Comment:
Do we need `pulp` for `tobler`?

Reply:
no; its all commented out, but i'll remove it anyway